chsdi/views/files.py

```diff
@@ -0,0 +1,242 @@
+            'Access-Control-Allow-Credentials': 'true'})
+        return ''
+
+    @view_config(route_name='files_collection', request_method='POST')
+    @requires_authorization()
+    @validate_kml_input()
+    def create_file(self):
+        self.file_id = self._get_uuid()
+        self.admin_id = self._get_uuid()
+        mime = self.request.content_type
+        data = self.request.body
+        try:
+            k = Key(self.bucket)
+            k.key = self.file_id
+            k.set_metadata('Content-Type', mime)
+            k.set_contents_from_string(data, replace=False)
+            _add_item(self.admin_id, self.file_id)
+
+            return {'adminId': self.admin_id, 'fileId': self.file_id}
+        except:
+            raise exc.HTTPInternalServerError('Cannot create file')
+
+    @view_config(request_method='GET')
+    def read_file(self):
+        try:
+            if self.admin_id is not None:
+                return {'fileId': self.file_id}
+            data = self.key.get_contents_as_string()
+            response = Response(data, content_type=self.key.content_type)
+            return response
+        except:
+            raise exc.HTTPNotFound('File %s not found' % self.file_id)
+
+    @view_config(request_method='POST')
+    @requires_authorization()
+    @validate_kml_input()
+    def update_file(self):
+        data = self.request.body
+        if self.admin_id is not None:
+            try:
+                self.key.set_contents_from_string(data, replace=True)
+                return {'adminId': self.admin_id, 'fileId': self.file_id, 'status': 'updated'}
+            except:
+                raise exc.HTTPInternalServerError('Cannot save file')
+        else:
+            # Fork file, get new file ids
+            # Duplicate of create_file function
+            self.file_id = self._get_uuid()
+            self.admin_id = self._get_uuid()
+            mime = self.request.content_type
+            data = self.request.body
+            try:
+                k = Key(self.bucket)
+                k.key = self.file_id
+                k.set_metadata('Content-Type', mime)
+                k.set_contents_from_string(data, replace=False)
+                _add_item(self.admin_id, self.file_id)
+
+                return {'adminId': self.admin_id, 'fileId': self.file_id, 'status': 'copied'}
+            except:
+                raise exc.HTTPInternalServerError('Cannot create file')
+
+    @view_config(request_method='DELETE')
+    @requires_authorization()
+    def delete_file(self):
+        try:
+            if self.admin_id is not None:
+                self.bucket.delete_key(self.key)
+                return Response(body={'success': True}, status='200 OK')
+            else:
+                raise exc.HTTPInternalServerError('Cannot delete file %s' % self.file_id)
+        except:
+            raise exc.HTTPInternalServerError('Cannot delete file %s' % self.file_id)
```

Comment:
I would rewrite that like this:

    if self.admin_id is None:
        raise exc.HTTPInternalServerError('Cannot delete file %s' % self.file_id)
    else:
        try:
            self.bucket.delete_key(self.key)
            Response(body={'success': True}, status='200 OK')
        except:
             exc.HTTPInternalServerError('Cannot delete file %s' % self.file_id)